### PR TITLE
Add a test case for serialization of a complex XML

### PR DESCRIPTION
**Pull request by arukshani**
$subject

## Check List 
- [ ] Read the [Contributing Guide](https://github.com/ballerina-platform/ballerina-lang/blob/master/CONTRIBUTING.md)
- [ ] Required Balo version update
- [ ] Updated Change Log
- [ ] Checked Tooling Support (#<Issue Number>)
- [ ] Added necessary tests
  - [ ] Unit Tests
  - [ ] Spec Conformance Tests
  - [ ] Integration Tests
  - [ ] Ballerina By Example Tests
- [ ] Increased Test Coverage   
- [ ] Added necessary documentation  
  - [ ] API documentation 
  - [ ] Module documentation in Module.md files
  - [ ] Ballerina By Examples

### CODE REVIEW

File: tests/jballerina-integration-test/src/test/java/org/ballerinalang/test/service/http/sample/SerializeComplexXmlTestCase.java [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+
+import static org.ballerinalang.jvm.XMLFactory.isEqual;
+import static org.ballerinalang.jvm.XMLFactory.parse;
+
+/**
+ * Test case for XML Serialization.
+ */
+@Test(groups = "http-test")
+public class SerializeComplexXmlTestCase extends HttpBaseTest {
+    private static final String XML_DATA = "<?xml version=\"1.0\"?>\n" +
```
Comment (SupunS) [inline tests/jballerina-integration-test/src/test/java/org/ballerinalang/test/service/http/sample/SerializeComplexXmlTestCase.java]: Would it be better to read from the `ComplexTestXmlSample.xml` itself for the assertion as well?
So we don't have to duplicate the same XML in two places

File: tests/jballerina-integration-test/src/test/resources/http/ComplexTestXmlSample.xml [hunk 2]
```diff
@@ -0,0 +1,50 @@
+<?xml version="1.0"?>
```
Comment (riyafa) [inline tests/jballerina-integration-test/src/test/resources/http/ComplexTestXmlSample.xml]: Shall we add copyright header?

File: tests/jballerina-integration-test/src/test/java/org/ballerinalang/test/service/http/sample/SerializeComplexXmlTestCase.java (hunk 1 above)
Comment (riyafa) [inline tests/jballerina-integration-test/src/test/java/org/ballerinalang/test/service/http/sample/SerializeComplexXmlTestCase.java]: Can't we move this data also to a file?

File: tests/jballerina-integration-test/src/test/resources/http/src/httpservices/41_serialize_xml.bal [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+    @http:ResourceConfig {
+        methods:["GET"],
+        path:"/xml"
+    }
+    resource function serializeXml (http:Caller caller, http:Request req) {
+        //Create an `xml` body part as a file upload.
+        mime:Entity xmlFilePart = new;
+        xmlFilePart.setContentDisposition(
+                       getContentDispositionForFormData("xml file part"));
+        xmlFilePart.setFileAsEntityBody("../../../src/test/resources/http/ComplexTestXmlSample.xml",
```
Comment (riyafa) [inline tests/jballerina-integration-test/src/test/resources/http/src/httpservices/41_serialize_xml.bal]: Shouldn't we use the path api for this? This path might be machine dependant

Comment (arukshani) [reply tests/jballerina-integration-test/src/test/resources/http/src/httpservices/41_serialize_xml.bal]: this is resolved internally by io APIs

**Pull request merged**